### DISCUSSION TITLE
System setup guide - best practices

### DISCUSSION THREAD

**Discussion by tesseract76** [esnet/iperf#1334] [Q&A]
Hello,

I am completely new to iPerf and I am looking for guidance on the correct or best system setup for the following scenario...hopefully recognizing the hardware specifications and noting if they require changes.

Server : Ubuntu 18.04 ; iPerf3 ; 64 GB ram ; 10 GB interface card
Clients: 25 wifi based Android tablets

How should iPerf3 be configured on the server (ie systemd? spin up threads on startup?)

If a connection between a tablet and the server drops but comes back does the iPerf3 startup again or is there a reconnect script or command that can be used at server startup that can be used?

Wanting to optimize the traffic to each tablet and based on the 10 GB interface card should we over subscribe the amount of data rate?

Should the server be upgraded to the latest Ubuntu release

Any help with this setup would be appreciated along with any other considerations that would help us get started.

regards,

**Comment by bmah888** [comment]
> I am completely new to iPerf and I am looking for guidance on the correct or best system setup for the following scenario...hopefully recognizing the hardware specifications and noting if they require changes.
> 
> Server : Ubuntu 18.04 ; iPerf3 ; 64 GB ram ; 10 GB interface card Clients: 25 wifi based Android tablets
> 
> How should iPerf3 be configured on the server (ie systemd? spin up threads on startup?)

It's not clear to me what you want to measure and/or test. That can actually be fairly important information, as it might determine whether iperf3 is even a suitable tool for your project.

> If a connection between a tablet and the server drops but comes back does the iPerf3 startup again or is there a reconnect script or command that can be used at server startup that can be used?

iperf3 tests are intended to run for a certain amount of time (or until a certain amount of data has been transferred). If a client and server process lose communication, they will continue to retry communications within the limits of the TCP protocol (or whatever transport protocol you have selected for the tests). Beyond that the test will just fail and/or terminate.

> Wanting to optimize the traffic to each tablet and based on the 10 GB interface card should we over subscribe the amount of data rate?

It sounds like you want the 25 tablets to all connect to the server simultaneously? Note that the iperf3 server process only supports a connection from a single client process at a time...this is by design to avoid tests interfering with each other. If you want to measure how you can send data from the server to all 25 tablets, iperf3 is probably not the best solution for you.

> Should the server be upgraded to the latest Ubuntu release

For iperf3 it probably doesn't matter much.

Good luck!
